### PR TITLE
[stable/nginx-ingress] fix servicemonitor selector

### PR DESCRIPTION
**Pull request by dennybaa**

### CODE REVIEW

**Comment by ChiefAlexander** [comment]
The selector in the deployment is immutable and thus this change would break anyone using the chart.

**Comment by dennybaa** [comment]
Hey @ChiefAlexander. Thank you, that's right. No necessity to change the selector.

**Comment by dennybaa** [comment]
Also I want to point out I tried to fix the serviceMonitor which has had a regression in #21361 by fixing it use the new `app.kubernetes.io/component`. However either prometheus operator doesn't propagate the dot-slashed value or prometheus process it correctly. So the old `componet` has to be preserved.

**Comment by ChiefAlexander** [comment]
Could you also readd the label to the daemonset and default backend as well?

**Comment by dennybaa** [comment]
@ChiefAlexander Yup will do. Daemonset should be changed too. But what do you think about the default-backend*, since there's no service monitor for that there's also no strong necessity to bring `component: "{{ .Values.defaultBackend.name }}"` back. ?

**Comment by ChiefAlexander** [comment]
/ok-to-test
Sounds good to me.

**Comment by dennybaa** [comment]
@ChiefAlexander I've added the change into the daemonset. please ok-to-test again

File: stable/nginx-ingress/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: nginx-ingress
-version: 1.36.0
+version: 1.36.1
```
Comment (ChiefAlexander) [inline stable/nginx-ingress/Chart.yaml]:
```suggestion
version: 1.36.2
```

**Comment by ChiefAlexander** [comment]
Thank for you fixing this!

**Comment by dennybaa** [comment]
welcome!

**Pull request merged**